src/components/MonitorListFilter.vue

```diff
@@ -191,52 +194,63 @@ export default {
+                Array.from(Object.entries(this.$router.currentRoute.value.query ?? {}))
+            );
+
+            return {
+                status: filters["status"] ? filters["status"].split(",") : [],
+                active: filters["active"] ? filters["active"].split(",") : [],
+                tags: filters["tags"] ? filters["tags"].split(",") : [],
+            };
+        },
         toggleStatusFilter(status) {
             let newFilter = {
-                ...this.filterState
+                ...this.getActiveFilters(),
+            };
+
+            const statusStates = {
+                1: "up",
+                0: "down",
+                2: "pending",
+                3: "maintenance",
             };
 
-            if (newFilter.status == null) {
-                newFilter.status = [ status ];
+            const finalStatus = statusStates[status];
+
+            if (newFilter.status.includes("" + finalStatus)) {
+                newFilter.status = newFilter.status.filter(item => item !== "" + finalStatus);
             } else {
-                if (newFilter.status.includes(status)) {
-                    newFilter.status = newFilter.status.filter(item => item !== status);
-                } else {
-                    newFilter.status.push(status);
-                }
+                newFilter.status.push(finalStatus);
             }
+
             this.$emit("updateFilter", newFilter);
         },
         toggleActiveFilter(active) {
             let newFilter = {
-                ...this.filterState
+                ...this.getActiveFilters(),
             };
 
-            if (newFilter.active == null) {
-                newFilter.active = [ active ];
+            if (newFilter.active.includes("" + active)) {
+                newFilter.active = newFilter.active.filter(item => item !== "" + active);
             } else {
-                if (newFilter.active.includes(active)) {
-                    newFilter.active = newFilter.active.filter(item => item !== active);
-                } else {
-                    newFilter.active.push(active);
-                }
+                newFilter.active.push(active);
             }
+
             this.$emit("updateFilter", newFilter);
         },
         toggleTagFilter(tag) {
             let newFilter = {
-                ...this.filterState
+                ...this.getActiveFilters(),
             };
 
-            if (newFilter.tags == null) {
-                newFilter.tags = [ tag.id ];
+            if (newFilter.tags.includes("" + tag.name)) {
+                newFilter.tags = newFilter.tags.filter(item => item !== "" + tag.name);
             } else {
-                if (newFilter.tags.includes(tag.id)) {
-                    newFilter.tags = newFilter.tags.filter(item => item !== tag.id);
-                } else {
-                    newFilter.tags.push(tag.id);
-                }
+                newFilter.tags.push(tag.name);
```

Comment:
This looks like it changes the filtering logic. Is ths change correct?

Reply:
Yes. The change is correct and I tested it. I am converting the boolean things to string (e.g. ''+true === 'true') when comparing them because the `active` value in the URL is a "stringified boolean".

Although it works, stringifying values shouldn't have been the case for `toggleTagFilter` and `toggleStatusFilter` - only needed it for `toggleActiveFilter` to compare the passed values with the values in the URL. Thanks for the catch, will fix it in the next commit.